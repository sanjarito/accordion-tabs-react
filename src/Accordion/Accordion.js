import React from 'react'

class Accordion extends React.Component {

  static defaultProps = {
    sections:[]

  }

  constructor(props){
    console.log('props in constructor', props)
    super(props)
    this.state = {
     activeSectionIndex:null,
    }
  }

  handleButtonClick(index){
    this.setState({
      activeSectionIndex:index
    })
    console.log('button clicked' + index)
  }

  renderContent(index){
    console.log(this.state.activeSectionIndex)
    console.log(index)


    const currentTab = this.props.sections[this.state.activeSectionIndex]
    return (
      <p className='content'>
        {currentTab.content}
      </p>
    )
  }


render(){
  return(
    <ul>
      {this.props.sections.map((prop,index)=>(
      <li>
      <button key={index} onClick={() => this.handleButtonClick(index)}>{prop.title}
      </button>
      {(this.state.activeSectionIndex === index && !!this.props.sections.length) && this.renderContent(index)}
      </li>
      ))}

    </ul>
  )
}
}

export default Accordion
