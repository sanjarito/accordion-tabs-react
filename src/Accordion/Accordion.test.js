import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Accordion from './Accordion';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const sections = [
  {
    title: 'Section 1',
    content: 'This is the content of section 1',
  },
  {
    title: 'Section 2',
    content: 'SECTION 2 CONTENT IS RIGHT HERE BOOOOYAHH',
  },
  {
    title: 'Section 3',
    content: 'aNDDDD SECTION 3 GOESSS  HEREEEE',
  },
]

describe(`Accordion Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('The component renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens a clicked section', () => {
  const wrapper = shallow(<Accordion sections={sections} />)
  wrapper.find('button').at(1).simulate('click')
  expect(toJson(wrapper)).toMatchSnapshot()
})


  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
})


})
