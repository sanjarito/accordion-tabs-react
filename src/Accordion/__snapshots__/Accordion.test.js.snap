// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component The component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component only opens one section at a time 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p
      className="content"
    >
      aNDDDD SECTION 3 GOESSS  HEREEEE
    </p>
  </li>
</ul>
`;

exports[`Accordion Component opens a clicked section 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p
      className="content"
    >
      SECTION 2 CONTENT IS RIGHT HERE BOOOOYAHH
    </p>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no tabs 1`] = `<ul />`;

exports[`Accordion Component renders the first tab by default 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
